include/contractor/files.hpp

```diff
@@ -0,0 +1,46 @@
+#ifndef OSRM_CONTRACTOR_FILES_HPP
+#define OSRM_CONTRACTOR_FILES_HPP
+
+#include "contractor/query_graph.hpp"
+
+#include "util/serialization.hpp"
+
+#include "storage/io.hpp"
+
+namespace osrm
+{
+namespace contractor
+{
+namespace files
+{
+
+// reads .osrm.hsgr file
+template <bool UseSharedMemory>
```

Comment:
`UseSharedMemory` is non relevant detail here, better to make `template<typename QueryGrahp>` in this PR